flink-runtime/src/test/java/org/apache/flink/runtime/dispatcher/MemoryExecutionGraphInfoStoreTest.java

```diff
@@ -0,0 +1,288 @@
+                    executionGraphInfoStore.getStoredJobsOverview(),
+                    Matchers.equalTo(expectedJobsOverview));
+        }
+    }
+
+    /** Tests that we obtain the correct collection of available job details. */
+    @Test
+    public void testAvailableJobDetails() throws IOException {
+        final int numberExecutionGraphs = 10;
+        final Collection<ExecutionGraphInfo> executionGraphInfos =
+                generateTerminalExecutionGraphInfos(numberExecutionGraphs);
+
+        final Collection<JobDetails> jobDetails = generateJobDetails(executionGraphInfos);
+
+        try (final MemoryExecutionGraphInfoStore executionGraphInfoStore =
+                createMemoryExecutionGraphInfoStore()) {
+            for (ExecutionGraphInfo executionGraphInfo : executionGraphInfos) {
+                executionGraphInfoStore.put(executionGraphInfo);
+            }
+
+            assertThat(
+                    executionGraphInfoStore.getAvailableJobDetails(),
+                    Matchers.containsInAnyOrder(jobDetails.toArray()));
+        }
+    }
+
+    /** Tests that an expired execution graph is removed from the execution graph store. */
+    @Test
+    public void testExecutionGraphExpiration() throws Exception {
+        final Time expirationTime = Time.milliseconds(1L);
+
+        final ManuallyTriggeredScheduledExecutor scheduledExecutor =
+                new ManuallyTriggeredScheduledExecutor();
+
+        final ManualTicker manualTicker = new ManualTicker();
+
+        try (final MemoryExecutionGraphInfoStore executionGraphInfoStore =
+                new MemoryExecutionGraphInfoStore(
+                        expirationTime, Integer.MAX_VALUE, scheduledExecutor, manualTicker)) {
+
+            final ExecutionGraphInfo executionGraphInfo =
+                    new ExecutionGraphInfo(
+                            new ArchivedExecutionGraphBuilder()
+                                    .setState(JobStatus.FINISHED)
+                                    .build());
+
+            executionGraphInfoStore.put(executionGraphInfo);
+
+            // there should one execution graph
+            assertThat(executionGraphInfoStore.size(), Matchers.equalTo(1));
+
+            manualTicker.advanceTime(expirationTime.toMilliseconds(), TimeUnit.MILLISECONDS);
+
+            // this should trigger the cleanup after expiration
+            scheduledExecutor.triggerScheduledTasks();
+
+            assertThat(executionGraphInfoStore.size(), Matchers.equalTo(0));
+
+            assertThat(
+                    executionGraphInfoStore.get(executionGraphInfo.getJobId()),
+                    Matchers.nullValue());
+
+            // check that the persisted file has been deleted
+            assertThat(executionGraphInfoStore.size(), Matchers.equalTo(0));
+        }
+    }
+
+    /** Tests that all persisted files are cleaned up after closing the store. */
+    @Test
+    public void testCloseCleansUp() throws IOException {
```

Comment:
Better to use `try` clause here in case there is `IOException` from `Store#put`.